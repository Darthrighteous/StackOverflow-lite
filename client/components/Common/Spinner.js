import React from 'react';
import PropTypes from 'prop-types';

const Spinner = (props) => {
  const { isLoading } = props;
  return (
    (isLoading) ? (
      <div className="spinner-overlay">
        <i className="fas fa-spinner fa-spin" />
      </div>
    ) : (null)
  );
};

Spinner.propTypes = {
  isLoading: PropTypes.bool.isRequired
};

export default Spinner;

