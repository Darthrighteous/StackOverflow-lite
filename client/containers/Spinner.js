import { connect } from 'react-redux';

import Spinner from '../components/Common/Spinner';

const mapStateToProps = state => ({
  isLoading: state.global.isLoading,
});

export default connect(mapStateToProps)(Spinner);