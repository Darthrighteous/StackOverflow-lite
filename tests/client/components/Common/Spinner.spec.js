import React from 'react';
import { shallow } from 'enzyme';

import Spinner from '../../../../client/components/Common/Spinner';

describe('Spinner snapshot tests', () => {
  test('not loading', () => {
    const spinner = shallow(
      <Spinner isLoading={false} />
    );
    expect(spinner).toMatchSnapshot();
  });

  test('is loading', () => {
    const spinner = shallow(
      <Spinner isLoading />
    );
    expect(spinner).toMatchSnapshot();
  });
});
