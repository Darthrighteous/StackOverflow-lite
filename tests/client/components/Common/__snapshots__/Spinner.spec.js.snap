// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner snapshot tests is loading 1`] = `
<div
  className="spinner-overlay"
>
  <i
    className="fas fa-spinner fa-spin"
  />
</div>
`;

exports[`Spinner snapshot tests not loading 1`] = `""`;
